documentation/specs/proposed/RAR_caching_onepager.md

```diff
@@ -0,0 +1,54 @@
+## RAR caching
+
+During every build we need to gather the graph of references and pass
+them to the compiler. This process is highly cacheable as the references
+don’t change all that often. Currently we have some limited caching in
+place however the way nodes are assigned work results in frequent cache
+misses.
+
+### Goals and motivations
+
+1ES team wants to isolate their File I/O related to the which is causing
+issues to their debugging efforts.
+
+Our motivation is a possible performance gain however we’re fine with
+the change as long as the impact is not negative.
+
+### Impact
+
+The only impact we’re concerned about is the performance. There will be
```

Comment:
Who will benefit from this feature? 1ES only or external customers as well?

Reply:
This mostly depends on the performance impact. If the performance is neutral, it will be mostly 1ES who will benefit (even though I imagine there can be a customer who would benefit in a similar way to the 1ES)
If the performance is positive, then it could have a broader impact. 

Also, one of our discussion items was to transition from multiprocess to multithreaded - if we can pull that one off, this will lose the burden of IPC, further improving the perf. However that is more of a "wishlist/longerm" sort of plan.